app/knight/knight.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(
+            self,
+            knight_dict: dict
+    ) -> None:
+        self.name = knight_dict.get("name")
+        self.power = knight_dict.get("power")
+        self.hp = knight_dict.get("hp")
+        self.weapon = knight_dict.get("weapon")
+        self.armour = knight_dict.get("armour")
+        self.potion = knight_dict.get("potion")
+        self.battle_ready_stats = self.battle_ready_stats()
+
+    # calculates battle ready knight's stats
+    def battle_ready_stats(self) -> dict:
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += self.potion["effect"]["power"]
+            if "hp" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["hp"]
+            if "protection" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["protection"]
+
+        # battle_hp -> sum of hp, armour protection ...
+        # ... and(if) potion hp or/and protection
+        battle_hp = self.hp + \

```

Comment:
There is no need for the `battle_hp` variable, you can increment `self.hp` directly.